src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,148 @@
+import os
+import re
+import sys
+from dataclasses import dataclass
+from functools import lru_cache
+from textwrap import dedent
+
+from saturn_engine.worker.executors.bootstrap import RemoteException
+
+
+@dataclass
+class ExceptionDetails:
+    message: str
+    exception_type: Exception
+    module: str
+    lineno: int
+
+
+@dataclass
+class ExceptionFilter(ExceptionDetails):
+    action: str
+
+
+def resolve_exception(exception_type: str) -> Exception:
+    if "." not in exception_type:
+        import builtins as m
+
+        klass = exception_type
+    else:
+        module, _, klass = exception_type.rpartition(".")
+        m = __import__(module, None, None, [klass])
+    return getattr(m, klass)
+
+
+def does_output_match(
```

Comment:
Nothing output specific here. I would suggest `does_error_match` and use keyword argument when there's no natural order (Add `*` to make them mandatory kwonly).